3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -8,18 +8,16 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
-
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
+      }
```

Comment:
```suggestion
      if (!firstName) {
        reject(new Error("You didn't pass in a first name!"));
        return;
      }
```

In this case, once the promise is rejected there is no point to continue with the rest of the function.  Therefore it is best to return immediately and skip the pointless call to `resolve()`.

Reply:
Thank you Jim. I updated my code as you mentioned.